src/modules/IsnsServer.rb

```diff
@@ -351,7 +351,9 @@ def isnsadm(params, ret_result = false)
 
     def isnsadm_control
       if !@isctrlnode
-        @isctrlnode = isnsadm("--register control")
+        res = isnsadm("--register control", ret_result = true)
```

Comment:
Why not just `isnsadm("--register control", true)`? `ret_result` is a variable that you're not going to use anymore.